src/session-props.ts

```diff
@@ -11,28 +11,35 @@ import { _info } from './utils/event-utils'
 import { SessionIdManager } from './sessionid'
 import { PostHogPersistence } from './posthog-persistence'
 import { CLIENT_SESSION_PROPS } from './constants'
+import { _strip_empty_properties } from './utils'
 
-interface SessionSourceProps {
-    initialPathName: string
-    referringDomain: string // Is actually host, but named domain for internal consistency. Should contain a port if there is one.
-    utm_medium?: string
-    utm_source?: string
-    utm_campaign?: string
-    utm_content?: string
-    utm_term?: string
+// this might be stored in a cookie with a hard 4096 byte limit, so save characters on key names
+export interface SessionSourceProps {
+    p?: string // initial pathname
+    r?: string // referring domain
+    m?: string // utm medium
+    s?: string // utm source
+    c?: string // utm campaign
+    n?: string // utm content
+    t?: string // utm term
 }
 
-interface StoredSessionSourceProps {
-    sessionId: string
-    props: SessionSourceProps
+export interface StoredSessionSourceProps {
+    s: string // session id
+    p: SessionSourceProps
 }
 
 export const generateSessionSourceParams = (): SessionSourceProps => {
-    return {
-        initialPathName: window.location.pathname,
-        referringDomain: _info.referringDomain(),
-        ..._info.campaignParams(),
-    }
+    const campaignParams = _info.campaignParams()
+    return _strip_empty_properties({
+        p: window.location.pathname,
+        r: _info.referringDomain(),
+        m: campaignParams.utm_medium,
+        s: campaignParams.utm_source,
+        c: campaignParams.utm_campaign,
+        n: campaignParams.utm_content,
+        t: campaignParams.utm_term,
```

Comment:
This won't actually help that much... 4Kb is not a lot especially as urls and things will be pretty long.
The main reason we see this issue on posthog.com is that we are not using `localstorage+cookie` persistence so I think this error can happen a lot anyways.

Reply:
Every little helps 🤷  - but yeah the main thing is using the right storage engine, and the other main thing is not storing values I don't need

Reply:
fly-by without having read or thought about the change much...

It would be nice to have the saving in the cookie without needing to read the tiny identifiers when being a human looking at the code. Feels easy to break something because there's not much for your brain to hold on to when reading it.